src/components/AttachmentList/AttachmentList.jsx

```diff
@@ -1,148 +1,124 @@
-
-  handleApplyFade = () => {}
-  handleScrollToEnd = () => {}
-
-  componentDidMount() {
-    this.handleApplyFade()
-  }
-
-  componentDidUpdate(prevProps) {
-    if (this.didAddNewAttachment(prevProps)) {
-      this.handleOnAddNewAttachment()
+export const AttachmentList = props => {
+  const {
+    children,
+    className,
+    downloadAllLabel,
+    onDownloadAllClick,
+    showDownloadAll,
+    withOverflow,
+    onScrollEnd,
+    theme: themeProp,
+    ...rest
+  } = props
+
+  const attachmentChildren = React.Children.toArray(children).filter(
+    child => child.type && child.type === Attachment
+  )
+
+  const { theme: themeContext } = useContext(AttachmentContext) || {}
+
+  const overflowCallback = useRef({
+    handleApplyFade: noop,
+    handleScrollToEnd: noop,
+  })
+
+  const theme = themeContext || themeProp
+  const isThemePreview = theme === 'preview'
+
+  useEffect(() => {
+    overflowCallback.current.handleApplyFade()
+    overflowCallback.current.handleScrollToEnd()
+    onScrollEnd && onScrollEnd()
+  }, [attachmentChildren.length, onScrollEnd])
+
+  const componentClassName = classNames(
+    'c-AttachmentList',
+    theme && `is-theme-${theme}`,
+    withOverflow && 'withOverflow',
+    className
+  )
+
+  const getDownloadAllMarkup = () => {
+    if (!showDownloadAll || attachmentChildren.length <= 1) return null
+
+    if (isThemePreview) {
+      return (
+        <IconButton
+          className="AttachmentList__DownloadAll"
+          theme="grey"
+          icon="inbox"
+          seamless
+          onClick={onDownloadAllClick}
+          title={downloadAllLabel}
+        />
+      )
     }
-  }
-
-  /**
-   * Method to update the UI (DOM) when new attachments have been added.
-   */
-  handleOnAddNewAttachment() {
-    this.handleScrollToEnd()
-  }
-
-  /**
-   * Determines if new attachments have been added.
-   *
-   * @param   {Object} prevProps The previous props.
-   * @returns {boolean} Whether new attachments have been added.
-   */
-  didAddNewAttachment(prevProps = this.props) {
-    if (!this.props.children) return false
-
-    const prevAttachments = React.Children.toArray(prevProps.children)
-    const currentAttachments = React.Children.toArray(this.props.children)
-
-    return prevAttachments.length < currentAttachments.length
-  }
 
-  render() {
-    const {
-      children,
-      className,
-      downloadAllLabel,
-      onDownloadAllClick,
-      showDownloadAll,
-      withOverflow,
-      ...rest
-    } = this.props
-    const { theme } = this.context
-
-    const isThemePreview = theme === 'preview'
-
-    const componentClassName = classNames(
-      'c-AttachmentList',
-      theme && `is-theme-${theme}`,
-      withOverflow && 'withOverflow',
-      className
-    )
-
-    const attachmentChildren = React.Children.toArray(children).filter(
-      child => child.type && child.type === Attachment
+    return (
+      <Attachment
+        elementClassName="AttachmentList__DownloadAll"
```

Comment:
should this be `c-AttachmentList__DownloadAll`? What're y'all's rules around prepending the classnames with `c-`?

Reply:
it should